### PR TITLE
feat(storage): Update list api documentation to add pagination examples

### PR DESCRIPTION
**Pull request by sdhuka**
#### Description of changes:
- Remove example for deprecated list API
- Add examples with `StoragePagedListOptions`

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [X] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [X] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [X] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/fragments/lib/storage/android/list.mdx [hunk 1]
```diff
@@ -1,14 +1,24 @@
-You can list all of the objects uploaded under a given prefix. This will list all public files:
+You can list all of the objects uploaded under a given prefix by setting the pageSize. If the page size is set lower than the total file size available,
```
Comment (dengdan154) [inline src/fragments/lib/storage/android/list.mdx:1]: You can list all of the objects uploaded under a given prefix by setting the pageSize. If the pagSize is set lower than the total file size available,

Comment (dengdan154) [inline src/fragments/lib/storage/android/list.mdx:1]:
```suggestion
You can list all of the objects uploaded under a given prefix by setting the pageSize. If the pageSize is set lower than the total file size available,
```

File: src/fragments/lib/storage/android/list.mdx [hunk 2]
```diff
@@ -18,11 +28,16 @@ Amplify.Storage.list("",
...
+val options = StoragePagedListOptions.builder()
+    .setPageSize(1000)
+    .build()
+
+Amplify.Storage.list("", options,
     { result ->
         result.items.forEach { item ->
             Log.i("MyAmplifyApp", "Item: ${item.key}")
         }
+        Log.i("MyAmplifyApp", "Next Token: ${result.nextToken}");
```
Comment (dengdan154) [inline src/fragments/lib/storage/android/list.mdx:40]:
```suggestion
        Log.i("MyAmplifyApp", "Next Token: ${result.nextToken}")
```

File: src/fragments/lib/storage/android/list.mdx [hunk 3]
```diff
@@ -32,10 +47,16 @@ Amplify.Storage.list("",
...
+val options = StoragePagedListOptions.builder()
+    .setPageSize(1000)
+    .build()
+
 try {
-    Amplify.Storage.list("").items.forEach {
-        Log.i("MyAmplifyApp", "Item: ${it.key}")
+    val result = Amplify.Storage.list("", options)
+    result.items.forEach {
+        Log.i("AmplifyApplication", "Item: $it")
```
Comment (dengdan154) [inline src/fragments/lib/storage/android/list.mdx:57]:
```suggestion
        Log.i("MyAmplifyApp", "Item: $it")
```

File: src/fragments/lib/storage/android/list.mdx [hunk 4]
```diff
@@ -32,10 +47,16 @@ Amplify.Storage.list("",
...
     }
+    Log.i("AmplifyApplication", "next token: ${result.nextToken}")
```
Comment (dengdan154) [inline src/fragments/lib/storage/android/list.mdx:59]:
```suggestion
    Log.i("MyAmplifyApp", "next token: ${result.nextToken}")
```

File: src/fragments/lib/storage/android/list.mdx [hunk 5]
```diff
@@ -45,29 +66,37 @@ try {
...
+RxAmplify.Storage.list("", options)
+.subscribe(
+    result -> {
+        for (StorageItem item : result.getItems()) {
+            Log.i("MyAmplifyApp", "Item: " + item.getKey());
+        }
+        Log.i("MyAmplifyApp", "Next Token: " + result.getNextToken());
+    },
+    error -> Log.e("MyAmplifyApp", "List failure", error)
+);
```
Comment (dengdan154) [inline src/fragments/lib/storage/android/list.mdx:82]: Tabbing is inconsistent

File: src/fragments/lib/storage/android/list.mdx [hunk 6]
```diff
@@ -123,20 +158,22 @@ try {
...
-        );
+.subscribe(
+    result -> {
+        for (StorageItem item : result.getItems()) {
+            Log.i("MyAmplifyApp", "Item: " + item.getKey());
+        }
+        Log.i("MyAmplifyApp", "Next Token: " + result.getNextToken());
+    },
+    error -> Log.e("MyAmplifyApp", "List failure", error)
+);
```
Comment (dengdan154) [inline src/fragments/lib/storage/android/list.mdx:176]: Tabbing is inconsistent

File: src/fragments/lib/storage/android/list.mdx [hunk 7]
```diff
@@ -105,15 +137,18 @@ Amplify.Storage.list("", options,
...
     .accessLevel(StorageAccessLevel.PROTECTED)
     .targetIdentityId("otherUserID")
+    .setPageSize(1000)
     .build()
 
 try {
-    Amplify.Storage.list("", options).items.forEach {
+    val result = Amplify.Storage.list("", options)
+    result.items.forEach {
         Log.i("AmplifyApplication", "Item: $it")
```
Comment (dengdan154) [inline src/fragments/lib/storage/android/list.mdx:149]:
```suggestion
        Log.i("MyAmplifyApp", "Item: $it")
```

File: src/fragments/lib/storage/android/list.mdx [hunk 8]
```diff
@@ -105,15 +137,18 @@ Amplify.Storage.list("", options,
...
     }
+    Log.i("AmplifyApplication", "next token: ${result.nextToken}")
```
Comment (dengdan154) [inline src/fragments/lib/storage/android/list.mdx:151]:
```suggestion
    Log.i("MyAmplifyApp", "next token: ${result.nextToken}")
```

**Review by dengdan154 — CHANGES_REQUESTED** [review]

**Review by dengdan154 — APPROVED** [review]

**Comment by sdhuka** [comment]
> @sdhuka, since a file was deleted from the `src/pages` and/or `src/fragments` directories, redirects might need to be set up so these previous pages do not 404. If redirects are needed, please answer these questions for each redirect that is needed:
> 
> * What is the source address of the redirect? (Where are you trying to redirect from?)
> * What is the target address of the redirect? (Where are you trying to redirect to?)
> * Type of redirect? 301 - permanent redirect or 302 - temporary redirect? (More info on Amplify Hosting redirects here: https://docs.aws.amazon.com/amplify/latest/userguide/redirects.html)

@timngyn @katiegoines No files are deleted in this PR, not sure why this bot picked it up?

**Review by hdworld11 — APPROVED** [review]

**Review by katiegoines — APPROVED** [review]

**Pull request merged**